src/openms/source/ANALYSIS/ID/AScore.cpp

```diff
@@ -458,10 +458,10 @@ namespace OpenMS
       
       tail = computePermutations_(tupel_left, tail_phospho_sites);
       
-      for (vector<vector<Size>>::iterator it = tail.begin(); it != tail.end(); ++it)
+      for (auto& t : tail)
       {
         vector<Size> temp(head);
-        temp.insert(temp.end(), it->begin(), it->end());
+        temp.insert(temp.end(), t.begin(), t.end());
         permutations.push_back(temp);
       }
```

Comment:
again, please take this an an optional suggestion:
the loop is rather inefficient, since it creates a temporary and does not move it.

(untested):
```
  for (const auto& t : tail)
  {
    auto& last = permutations.emplace_back(head);
    last.insert(last.end(), t.begin(), t.end());
  }
```